core/src/vault_registry.rs

```diff
@@ -0,0 +1,433 @@
+#[repr(C)]
+pub struct StMintEntry {
+    st_mint: Pubkey,
+    ncn_fee_group: NcnFeeGroup,
+    reward_multiplier_bps: PodU64,
+    switchboard_feed: Pubkey,
+    no_feed_weight: PodU128,
+    reserved: [u8; 32],
+}
+
+impl StMintEntry {
+    pub fn new(
+        st_mint: Pubkey,
+        ncn_fee_group: NcnFeeGroup,
+        reward_multiplier_bps: u64,
+        switchboard_feed: Pubkey,
+        no_feed_weight: u128,
+    ) -> Self {
+        Self {
+            st_mint,
+            ncn_fee_group,
+            reward_multiplier_bps: PodU64::from(reward_multiplier_bps),
+            switchboard_feed,
+            no_feed_weight: PodU128::from(no_feed_weight),
+            reserved: [0; 32],
+        }
+    }
+
+    pub fn no_feed_weight(&self) -> u128 {
+        self.no_feed_weight.into()
+    }
+
+    pub const fn st_mint(&self) -> Pubkey {
+        self.st_mint
+    }
+
+    pub const fn ncn_fee_group(&self) -> NcnFeeGroup {
+        self.ncn_fee_group
+    }
+
+    pub fn reward_multiplier_bps(&self) -> u64 {
+        self.reward_multiplier_bps.into()
+    }
+
+    pub const fn switchboard_feed(&self) -> Pubkey {
+        self.switchboard_feed
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.st_mint().eq(&Pubkey::default())
+    }
+}
+
+impl Default for StMintEntry {
+    fn default() -> Self {
+        Self::new(
+            Pubkey::default(),
+            NcnFeeGroup::default(),
+            0,
+            Pubkey::default(),
+            0,
+        )
+    }
+}
+
+#[derive(Debug, Clone, Copy, Zeroable, ShankType, Pod)]
+#[repr(C)]
+pub struct VaultEntry {
+    st_mint: Pubkey,
+    vault_index: PodU64,
+    reserved: [u8; 128],
+}
+
+impl VaultEntry {
+    pub fn new(mint: Pubkey, vault_index: u64) -> Self {
+        Self {
+            st_mint: mint,
+            vault_index: PodU64::from(vault_index),
+            reserved: [0; 128],
+        }
+    }
+
+    pub fn vault_index(&self) -> u64 {
+        self.vault_index.into()
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.vault_index() == u64::MAX
+    }
+}
+
+impl Default for VaultEntry {
+    fn default() -> Self {
+        Self::new(Pubkey::default(), u64::MAX)
+    }
+}
+
+#[derive(Debug, Clone, Copy, Zeroable, ShankType, Pod, AccountDeserialize, ShankAccount)]
+#[repr(C)]
+pub struct VaultRegistry {
+    pub ncn: Pubkey,
+    pub bump: u8,
+    pub reserved: [u8; 127],
+    pub st_mint_list: [StMintEntry; 64],
+    pub vault_list: [VaultEntry; 64],
+}
+
+impl Discriminator for VaultRegistry {
+    const DISCRIMINATOR: u8 = Discriminators::VaultRegistry as u8;
+}
+
+impl VaultRegistry {
+    pub const SIZE: usize = 8 + size_of::<Self>();
+
+    pub fn new(ncn: Pubkey, bump: u8) -> Self {
+        Self {
+            ncn,
+            bump,
+            reserved: [0; 127],
+            st_mint_list: [StMintEntry::default(); MAX_ST_MINTS],
+            vault_list: [VaultEntry::default(); MAX_VAULT_OPERATOR_DELEGATIONS],
```

Comment:
do you remember why we called this `MAX_VAULT_OPERATOR_DELEGATIONS` ? should just be `MAX_VAULTS` right?

Reply:
Correct, it should be MAX_VAULTS.